ci/pkl-pipelines/pay-dev/deploy-to-perf.pkl

```diff
@@ -0,0 +1,423 @@
+  new { name = "publicapi" }
+  new { name = "publicauth"; has_db = true;  rds_identifier_suffix = 1 }
+  new { name = "selfservice" }
+  new { name = "toolbox" }
+  new { name = "webhooks"; has_db = true }
+  new { name = "webhooks-egress"; include_nginx_and_adot_sidecar = false; use_app_specific_deploy_task = true }
+}.toList().sortBy((app) -> app.name).toListing()
+
+// sidecars or other helpers
+local payHelpers: Listing<PayApp> = new Listing<PayApp> {
+  new { name = "alpine"; override_app_to_deploy = "scheduled-tasks"; include_nginx_and_adot_sidecar = false }
+  new { name = "nginx-forward-proxy"; override_app_to_deploy = "frontend"; include_nginx_and_adot_sidecar = false }
+  new { name = "nginx-proxy"; override_ecr_repo = "docker-nginx-proxy"; override_app_to_deploy = "toolbox"; include_nginx_and_adot_sidecar = false }
+  new { name = "stream-s3-sqs"; override_app_to_deploy = "scheduled-tasks";  include_nginx_and_adot_sidecar = false }
+}.toList().sortBy((app) -> app.name).toListing()
+
+local allPayApps: Listing<PayApp> = new {
+  ...payApps
+  ...payHelpers
+}
+
+local payApplicationsWithDB = payApps.toList().filter((app) -> app.has_db).toListing()
+
+resource_types {
+  shared_resources_for_slack_notifications.slackNotificationResourceType
+  shared_resources_for_metrics.prometheusPushgatewayResourceType
+}
+
+resources = new {
+  pipeline_self_update.PayPipelineSelfUpdateResource("pay-dev/deploy-to-perf.pkl", "master")
+  shared_resources_for_metrics.prometheusPushgatewayResource
+  (shared_resources.payGithubResourceWithBranch("deploy-to-perf-pipeline-definition", "pay-ci", "master")) {
+    source { ["paths"] = new Listing<String> { "ci/pipelines/deploy-to-perf.yml" } }
+  }
+  shared_resources.payCiGitHubResource
+  shared_resources.payGithubResourceWithBranch("pay-infra", "pay-infra", "master")
+
+  shared_resources.payECRResourceWithVariant("adot-ecr-registry-perf", "govukpay/adot", "pay_aws_test_account_id", "perf")
+  for (app in allPayApps) {
+    shared_resources.payECRResourceWithVariant("\(app.name)-ecr-registry-perf", "govukpay/\(if (app.override_ecr_repo == null) app.name else app.override_ecr_repo)", "pay_aws_test_account_id", "perf")
+    when (app.has_db) {
+      shared_resources.payECRResourceWithVariant("\(app.name)-db-ecr-registry-perf", "govukpay/\(app.name)", "pay_aws_test_account_id", "perf-db")
+    }
+  }
+  shared_resources_for_slack_notifications.slackNotificationResource
+}
+
+groups {
+  for (app in allPayApps) {
+    new {
+      name = app.name
+      jobs = new Listing {
+        when (app.override_app_to_deploy != null) {
+          when (app.override_app_to_deploy == "scheduled-tasks") {
+            "deploy-\(app.override_app_to_deploy)"
+          } else {
+            "deploy-\(app.override_app_to_deploy)-to-perf"
+          }
+        } else {
+          "deploy-\(app.name)-to-perf"
+        }
+        when (app.has_db) {
+          "\(app.name)-db-migration-perf"
+        }
+      }
+    }
+  }
```

Comment:
You'll need to add the pipeline self update group, without it the pipeline wont apply since there will be a job which doesn't belong to any groups (which makes it impossible to see in the UI, thankfully the CLI gives an error for it, but only on the real set-pipeline, not in the dry-run apparently):

```suggestion
    pipeline_self_update.payPipelineSelfUpdateGroup
  }
```

Reply:
Added group